src/material/sidenav/drawer.ts

```diff
@@ -595,8 +595,9 @@ export class MatDrawer implements AfterViewInit, AfterContentChecked, OnDestroy
   /** Updates the enabled state of the focus trap. */
   private _updateFocusTrapState() {
     if (this._focusTrap) {
-      // The focus trap is only enabled when the drawer is open in any mode other than side.
-      this._focusTrap.enabled = this.opened && this.mode !== 'side';
```

Comment:
Mostly trying to understand the reasoning here: don't we still want to trap focus when the mode is `over`? In that case the sidenav basically acts as a dialog so I would expect that it needs focus trapping as well.

Reply:
I thing the thinking is that the backdrop enforces the modal-ness from the mouse perspective and the focus-trap enforces it from the keyboard perspective, so it makes sense to tie those together. That way we're presenting the same modal-ness to both mouse and keyboard users

Reply:
By default, the Drawer has a backdrop when the mode is `over`. Developer may override the default backdrop behavior using the `hasBackdrop` Input, so `over` mode will still have a backdrop unless the developer does `[hasBackdrop]="false"`.

^ That's right, Miles.

For accessibility, it doesn't make sense if you can interact with the mouse and not with the keyboard – or visa versa. We've had issue reports from developers that found focus trapping to be too aggressive when there is no backdrop (#https://github.com/angular/components/issues/26572).